backend/src/schema/resolvers/socialMedia.spec.js

```diff
@@ -24,92 +49,139 @@ describe('SocialMedia', () => {
+      beforeEach(async () => {
+        headers = await login(userParams)
+        client = new GraphQLClient(host, { headers })
+      })
+
+      it('creates social media with correct URL', async () => {
+        await expect(client.request(mutation, variables)).resolves.toEqual(
+          expect.objectContaining({
+            CreateSocialMedia: {
+              id: expect.any(String),
+              url: url,
+            },
+          }),
+        )
+      })
+
+      it('rejects empty string', async () => {
+        variables = { url: '' }
+
+        await expect(client.request(mutation, variables)).rejects.toThrow(
+          '"url" is not allowed to be empty',
+        )
+      })
+
+      it('rejects invalid URLs', async () => {
+        variables = { url: 'not-a-url' }
+
+        await expect(client.request(createSocialMediaMutation, variables)).rejects.toThrow(
+          '"url" must be a valid uri',
+        )
+      })
     })
   })
 
-  describe('authenticated', () => {
+  describe('update social media', () => {
     beforeEach(async () => {
-      headers = await login({
-        email: 'test@example.org',
-        password: '1234',
-      })
-      client = new GraphQLClient(host, {
-        headers,
+      headers = await login(ownerParams)
+      client = new GraphQLClient(host, { headers })
+
+      const { CreateSocialMedia } = await client.request(createSocialMediaMutation, { url })
+      const { id } = CreateSocialMedia
+
+      variables = { url: newUrl, id }
+      mutation = updateSocialMediaMutation
+    })
+
+    describe('unauthenticated', () => {
+      it('throws authorization error', async () => {
+        client = new GraphQLClient(host)
+        await expect(client.request(mutation, variables)).rejects.toThrow('Not Authorised')
       })
     })
 
-    it('creates social media with correct URL', async () => {
-      const variables = {
-        url: 'http://nsosp.org',
-      }
-      await expect(client.request(mutationC, variables)).resolves.toEqual(
-        expect.objectContaining({
-          CreateSocialMedia: {
-            id: expect.any(String),
-            url: 'http://nsosp.org',
-          },
-        }),
-      )
+    describe('authenticated as other user', () => {
+      // TODO: make sure it throws an authorization error
     })
 
-    it('deletes social media', async () => {
-      const creationVariables = {
-        url: 'http://nsosp.org',
-      }
-      const { CreateSocialMedia } = await client.request(mutationC, creationVariables)
+    describe('authenticated as owner', () => {
+      it('updates social media', async () => {
+        const expected = { UpdateSocialMedia: { ...variables } }
+
+        await expect(client.request(mutation, variables)).resolves.toEqual(
+          expect.objectContaining(expected),
+        )
+      })
+
+      describe('given a non-existent id', () => {
+        // TODO: make sure it throws an error
+      })
+    })
+  })
+
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-17T10:35:24Z" title="Wednesday, July 17th 2019, 12:35:24 pm +02:00">Jul 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -74,6 +82,27 @@ describe('SocialMedia', () => {
       )
     })
 
+    it('updates social media', async () => {
+      const creationVariables = {
+        url: 'http://nsosp.org',
+      }
+      const { CreateSocialMedia } = await client.request(mutationC, creationVariables)
+      const { id } = CreateSocialMedia
+      const variables = {
+        id,
+        url: 'https://newurl.org',
+      }
+      const expected = {
+        UpdateSocialMedia: {
+          id: id,
+          url: 'https://newurl.org',
+        },
+      }
+      await expect(client.request(mutationU, variables)).resolves.toEqual(
+        expect.objectContaining(expected),
+      )
+    })
+
```


It would be great to see some tests trying to update a socialMedia url without being authenticated/trying to change another's socialMedia url... it might be that the functionality already prevents these from happening, but with software tests if someone refactors it later, we can have the confidence it will still maintain those constraints

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-07-23T13:05:21Z" title="Tuesday, July 23rd 2019, 3:05:21 pm +02:00">Jul 23, 2019</time>_

---


I added a few, so I think it's an improvement already. :wink:

What I didn't manage is to test changing another user's links. I saw that you did something similar for `comments` and `posts` but I don't really understand what's going on with the different `factory` instances and I didn't just want to copy the code blindly...

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-07-25T10:11:38Z" title="Thursday, July 25th 2019, 12:11:38 pm +02:00">Jul 25, 2019</time>_

---


I just agreed with @roschaefer that I would refactor the social media backend (including the tests) in a separate `PR`.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-25T11:09:13Z" title="Thursday, July 25th 2019, 1:09:13 pm +02:00">Jul 25, 2019</time>_

---


That's great @alina-beck, I'm really happy you are finding your way, and doing such great work!